doc/lua_api.md

```diff
@@ -7815,6 +7821,10 @@ child will follow movement and rotation of that bone.
 
 ### Methods
 
+* `is_valid()`: returns whether the object is valid.
+   Objects may be invalidated either through explicit removal,
+   or implicitly between server steps by the engine.
+   **The below methods should only be called for valid objects.**
```

Comment:
also this explicitly makes any method calls on invalid objects UB. that includes many existing mods
do we want to do that?

IMO at the very least generic getters (e.g. `get_pos`) should be allowed to be called on invalid references

Reply:
I've checked the code and all getters consistently return nothing when called on invalid objects. It's probably fine to keep that for now, I wouldn't want to deprecate it yet - too much work for everyone involved.

(The "should" was intended as in "this is best practice if you're writing a mod for a recent Minetest version today", not as in "this is deprecated and will be broken soon". I have clarified.)